ga4gh/backend.py

```diff
@@ -384,6 +318,95 @@ def setMaxResponseLength(self, maxResponseLength):
         """
         self._maxResponseLength = maxResponseLength
 
+    # private methods used to implement readsGenerator
+
+    def _getPosition(self, readAlignment):
```

Comment:
These aren't really methods of the backend I think (``self isn't used, which is a good indication). I think it might be a good idea to make these as plain functions, and then pass these as parameters to your general methods for iterating over the intervals. I think it's probably worthwhile making a new class `IntervalIterator` to do this to avoid cluttering up the backend with the details of this. So, concretely, I'd image:
```python
def _getReadAlignmentStart(readAlignment):
    return blah blah

def _getReadAlignmentEnd(readAlignment):
    return blah blah

class ItervalPager(object):
    """
    Takes care of the details of paging over interval search results for reads and variant search.
    """
    def __init__(self, request, iterator, getStart, getEnd):
        self._request = request
        self._getStart = getStart
        # etc.
    # The rest of the interface is similar to what you have

"""

The advantage of this is, (1) we isolate the tricky interval paging logic into a single class, making it more easily testable; and (2) this should make it very easy to share code between the reads and variants search.

Does this make sense?